src/Lumi/Components2/Text.purs

```diff
@@ -0,0 +1,194 @@
+module Lumi.Components2.Text where
+
+import Prelude
+
+import Data.Lens (Setter')
+import Effect.Unsafe (unsafePerformEffect)
+import Lumi.Components (LumiComponent, LumiProps, PropsModifier, lumiComponent, propsModifier)
+import Lumi.Styles as S
+import Lumi.Styles.Theme (LumiTheme(..), useTheme)
+import React.Basic (JSX, ReactComponent)
+import React.Basic.DOM as R
+import React.Basic.Emotion as E
+import React.Basic.Hooks as Hooks
+
+-- Text
+
+data Text = Text
+
+type TextProps =
+  ( component :: Text
+  , element :: TextElement
+  , content :: String
+  )
+
+type TextElement = ReactComponent { children :: Array JSX, className :: String }
+
+text :: LumiComponent TextProps
+text =
+  unsafePerformEffect $ lumiComponent "Text" defaults \props -> Hooks.do
+    theme <- useTheme
+    pure $
+      E.element props.element
+        { children: [ R.text props.content ]
+        , className: props.className
+        , css: defaultTextStyle <> props.css theme
+        }
+  where
+    defaults :: Record TextProps
+    defaults =
+      { component: Text
+      , element: R.span'
+      , content: ""
+      }
+
+    defaultTextStyle =
+      S.css
+        { fontSize: S.str "14px"
+        , lineHeight: S.str "17px"
+        , margin: S.str "0"
+        , padding: S.str "0"
```

Comment:
I think we want `Text` to default to `white-space: pre;` and have `Paragraph` override that

Reply:
Maybe `pre-wrap` is safer?

Reply:
Maybe.. I was thinking we didn't want UI text/labels wrapping by default, but maybe not. With everything wrapping by default it's easy to forget, but having the Text/Paragraph distinction makes it obvious which is being chosen. _But_, wrapping is more common, so perhaps that'd be more annoying than helpful.

Reply:
I think you're right. I had been thinking we might want the base Text component to never wrap and Paragraph to contain the wrapping behavior/settings (like spacing/line-height which wouldn't be available on Text), but wrapping is just so much more common that that might be more annoying than helpful.

Reply:
Either way, Text and Paragraph should be setting it explicitly so we don't inherit unexpected values. We probably want to do the same with `font-family`, and pull both `font-family` and the sizing/line-height settings from the theme.

Reply:
Side note, if we did make Text nowrap and Paragraph wrap, we'd probably have trouble doing that with the current api, where Paragraph's children are Texts (or other components with Text children) as the inner Text nowrap would be more specific. That probably means we have the same issue with other styles here.. Paragraph's settings might need to be marked `important`, unless you can think of a better option.

Reply:
I see `paragraph_` is partially working around that [here](https://github.com/lumihq/purescript-lumi-components/pull/146/files#diff-eb2d5760b0fc747166b5c115e78c61f1R101), but `paragraph` would be dependent on each child Text having its own sizing/line-height set. Hmm. I think.. Paragraph defaults and overrides should override all child Text defaults, but not child Text overrides, i.e.
```purs
paragraph
  $ emphasize
  $$$
    -- paragraphs are not emphasized by default but this one has a modifier
    -- but this local override still takes priority
    [ text $ nonEmphasized $$$ "hello"
    -- paragraphs have default text size and line height, but this local override would still win
    -- this text would still appear emphasized though
    , text $ fontSize 20 $$$ "goodbye"
    ]
```

Reply:
These problems are kind of unique to Paragraph and Text

Reply:
> Either way, Text and Paragraph should be setting it explicitly so we don't inherit unexpected values. We probably want to do the same with font-family, and pull both font-family and the sizing/line-height settings from the theme.

Yes, excellent points!

> Side note, if we did make Text nowrap and Paragraph wrap

I don't think we should make Text nowrap. What would be the reason for doing that?

> Paragraph defaults and overrides should override all child Text defaults, but not child Text overrides, i.e.

Maybe we could get around that by having all `text` defaults be `inherit` and setting those defaults globally?

Reply:
> I don't think we should make Text nowrap. What would be the reason for doing that?

I think I agree, it was an idea I inherited from another api but it doesn't give us anything here.

> Maybe we could get around that by having all text defaults be inherit and setting those defaults globally? But then that'd kind of defeat the purpose of having LumiTheme :p

Maybe.. it might be risky if there are non-lumi-components styles used anywhere on the page